modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/upgrade/BlogsServiceUpgrade.java

```diff
@@ -95,12 +101,55 @@ public void register(Registry registry) {
 			new com.liferay.blogs.internal.upgrade.v2_1_1.UpgradeBlogsEntry());
 	}
 
+	private UnsafeFunction<String, Boolean, Exception>
+		_getUpgradeDiscussionSubscriptionClassNameUnsafeFunction() {
+
+		return className -> {
+			DynamicQuery groupDynamicQuery = _groupLocalService.dynamicQuery();
+
+			groupDynamicQuery.add(RestrictionsFactoryUtil.eq("site", true));
+
+			List<Long> groupIds = ListUtil.toList(
+				_groupLocalService.dynamicQuery(groupDynamicQuery),
+				Group.GROUP_ID_ACCESSOR);
+
+			DynamicQuery subscriptionDynamicQuery =
+				_subscriptionLocalService.dynamicQuery();
+
+			subscriptionDynamicQuery.add(
+				RestrictionsFactoryUtil.eq(
+					"classNameId",
+					_classNameLocalService.getClassNameId(className)));
+
+			List<Subscription> subscriptions =
+				_subscriptionLocalService.dynamicQuery(
+					subscriptionDynamicQuery);
+
+			for (Subscription subscription : subscriptions) {
+				if (groupIds.contains(subscription.getClassPK())) {
+					continue;
+				}
+
+				subscription.setClassName(
```

Comment:
@IstvanD @robertoDiaz

If the previous point is true we can create a method in UpgradeDiscussionSubscriptionClassName which receives a className and returns a function that does the upgrade className in subscriptions via SQL.

That would be much faster.

Reply:
It is not true 🤡